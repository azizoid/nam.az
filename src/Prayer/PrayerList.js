import React from "react";
import PropTypes from "prop-types";
import Prayer from "./Prayer";

const PrayerList = ({ prayers, currentPrayer }) => {
  return (
    <div className="row" id="times">
      {prayers.map((prayer, index) => {
        const classes = ["col-sm-12", "col-md-4", "alert"];
        let isCur = false;

        if (index === 1) {
          if (index === currentPrayer) {
            classes.push("col-lg-2");
          } else {
            classes.push("text-muted", "col-lg-1");
          }
        } else {
          if (index === currentPrayer) {
            isCur = true;
            classes.push("alert-success", "col-lg-3");
          } else {
            classes.push("col-lg-2");
          }
        }

        return (
          <Prayer
            classes={classes.join(" ")}
            prayer={prayer}
            current={isCur}
            index={index}
            key={prayer.id}
          />
        );
      })}
    </div>
  );
};

PrayerList.propTypes = {
  prayers: PropTypes.arrayOf(PropTypes.object).isRequired,
  currentPrayer: PropTypes.number,
};

export default PrayerList;
