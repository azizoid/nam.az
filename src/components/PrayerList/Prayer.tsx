import { MdBrightness7 } from 'react-icons/md';

export type PrayerProps = {
  id: number;
  title: string;
  time: string;
  rakat: number;
  ago: string;
};

export type PrayerContainerProps = {
  prayer: PrayerProps;
  classes: string;
  current: boolean;
  index: number;
};

export const Prayer = ({
  prayer,
  classes,
  current,
  index,
}: PrayerContainerProps): JSX.Element => (
  <div className={`prayer alert ${classes}`}>
    <h6>{prayer.title}</h6>

    <div className="flex flex-col">
      <h4 className="text-2xl">{prayer.time}</h4>

      <div className="flex justify-center">
        {!current ? (
          index !== 1 ? (
            <small>{prayer.ago}</small>
          ) : (
            <MdBrightness7 className="text-yellow-400" />
          )
        ) : (
          <span className="badge badge-yellow">
            Zaman varkən,
            <br />
            Namazını qıl!
          </span>
        )}
      </div>
    </div>
  </div>
);
