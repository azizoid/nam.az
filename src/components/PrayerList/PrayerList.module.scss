.times {
    text-align: center;
  
    h3 {
      font-weight: bold;
    }
    
    h6 {
      padding: 0;
    }
  }
  
  @media (max-width: 480px) {
    .times {
      text-align: center;
  
      .text-sm-right {
        text-align: right !important;
      }
  
      h2 {
        margin: 0;
      }
  
      .alert {
        margin-bottom: 0;
        padding-top: 0;
      }
    }
  }