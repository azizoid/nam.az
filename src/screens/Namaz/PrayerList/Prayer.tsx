import { MdBrightness7 } from 'react-icons/md'

export type PrayerProps = {
  id: number;
  title: string;
  time: string;
  rakat: number;
  ago: string;
};

export type PrayerContainerProps = {
  prayer: PrayerProps;
  classes: string;
  current: boolean;
  index: number;
};

export const Prayer = ({
  prayer: { title, time, ago },
  classes,
  current,
  index,
}: PrayerContainerProps) => (
  <div className={`prayer alert ${classes}`}>
    <h6 className="w-full text-right md:w-auto">{title}</h6>

    <div className="flex w-full flex-col md:w-auto">
      <h4 className="text-2xl">{time}</h4>

      <div className="flex justify-center">
        {!current ? (
          index === 1 ? (
            <MdBrightness7 className="text-yellow-400" />
          ) : (
            <small>{ago}</small>
          )
        ) : (
          <span className="badge badge-yellow">
            Zaman varkən,
            <br />
            Namazını qıl!
          </span>
        )}
      </div>
    </div>
  </div>
)
