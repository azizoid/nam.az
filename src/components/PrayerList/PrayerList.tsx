import { Prayer, PrayerProps } from './Prayer';

export type PrayerListProps = {
  prayers: PrayerProps[];
  currentPrayer?: number;
  progress?: number;
};

export const PrayerList = ({
  prayers,
  currentPrayer,
  progress = 0,
}: PrayerListProps): JSX.Element => (
  <div className="grid grid-cols-12 text-center">
    {prayers.map((prayer, index) => {
      let isCur = false;
      const finalClass = [];

      if (index === 1) {
        if (index !== currentPrayer) {
          finalClass.push('text-slate-400');
        }
      } else {
        if (index === currentPrayer) {
          isCur = true;
          finalClass.push(progress < 75 ? 'alert-success' : 'alert-danger');
        }
      }

      return (
        <Prayer
          classes={finalClass.join(' ')}
          prayer={prayer}
          current={isCur}
          index={index}
          key={index}
        />
      );
    })}
  </div>
);
