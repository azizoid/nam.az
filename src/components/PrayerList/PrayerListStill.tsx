import classNames from "classnames";
import { MdBrightness7 } from "react-icons/md";
import { PrayerListProps } from "./PrayerList";

import styles from './PrayerList.module.scss'

export const PrayerListStill = ({ prayers }: PrayerListProps):JSX.Element => (
  <div className={classNames("row", styles.times)}>
    {prayers.map((prayer, index) => (
        <div className="col-sm-12 col-md-4 alert alert-light text-muted" key={index}>
          <div className="row">
            <h6 className="col-7 col-md-12 align-self-center">{prayer.title}</h6>

            <div className="col-5 col-md-12">
              <h4>{prayer.time}</h4>

              {index === 1 && (
                <MdBrightness7  />
              )}
            </div>
          </div>
        </div>
      )
    )}
  </div>
);
