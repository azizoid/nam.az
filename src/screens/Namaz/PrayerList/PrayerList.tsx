import { Prayer, PrayerProps } from './Prayer'

export type PrayerListProps = {
  prayers: PrayerProps[];
  currentPrayer?: number;
};

export const PrayerList = ({ prayers, currentPrayer }: PrayerListProps) => (
  <div className="grid grid-cols-12 text-center">
    {prayers.map((prayer, index) => {
      const isCur = index === currentPrayer && index !== 1
      const classes = isCur ? 'alert-success' : index === 1 ? 'text-slate-400' : ''

      return (
        <Prayer
          classes={classes}
          prayer={prayer}
          current={isCur}
          index={index}
          key={prayer.id}
        />
      )
    })}
  </div>
)
