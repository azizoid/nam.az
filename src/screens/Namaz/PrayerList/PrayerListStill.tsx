import { MdBrightness7 } from 'react-icons/md'

import { PrayerListProps } from './PrayerList'

export const PrayerListStill = ({ prayers }: PrayerListProps) => (
  <div className="grid grid-cols-12 text-center">
    {prayers.map((prayer, index) => (
      <div className="prayer alert text-slate-400" key={index}>
        <h6>{prayer.title}</h6>

        <div className="flex flex-col">
          <h4 className="text-2xl">{prayer.time}</h4>

          <div className="flex justify-center">
            {index === 1 && <MdBrightness7 />}
          </div>
        </div>
      </div>
    ))}
  </div>
)
