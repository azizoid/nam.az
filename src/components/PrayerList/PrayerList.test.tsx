import { render } from '@testing-library/react';
import React from 'react';
import { PrayerList } from './PrayerList';

const mock = [
  { id: 1, title: 'Fəcr namazı', time: '-:-', rakat: 2, ago: '1 minute ago' },
  { id: 2, title: 'Günəş', time: '-:-', rakat: 0, ago: '2 minutes ago' },
  { id: 3, title: 'Zöhr namazı', time: '-:-', rakat: 4, ago: '3 minutes ago' },
  { id: 4, title: 'Əsr namazı', time: '-:-', rakat: 4, ago: '4 minutes ago' },
  {
    id: 5,
    title: 'Məğrib namazı',
    time: '-:-',
    rakat: 3,
    ago: '5 minutes ago',
  },
  { id: 6, title: 'İşa namazı', time: '-:-', rakat: 4, ago: '6 minutes ago' },
];

test('renders PrayerList component', async () => {
  const { container, findAllByText } = render(
    <PrayerList prayers={mock} currentPrayer={2} />
  );

  const prayers = await findAllByText(/namazı/);
  expect(prayers).toHaveLength(5);

  expect(container).toMatchSnapshot();
});
