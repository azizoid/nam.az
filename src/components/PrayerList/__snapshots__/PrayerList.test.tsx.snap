// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PrayerList component 1`] = `
<div>
  <div
    class="row times"
  >
    <div
      class="col-12 col-md-4 alert"
    >
      <div
        class="row"
      >
        <h6
          class="col-7 col-md-12 align-self-center"
        >
          Fəcr namazı
        </h6>
        <div
          class="col-5 col-md-12"
        >
          <h4>
            -:-
          </h4>
          <small>
            1 minute ago
          </small>
        </div>
      </div>
    </div>
    <div
      class="col-12 col-md-4 alert text-muted"
    >
      <div
        class="row"
      >
        <h6
          class="col-7 col-md-12 align-self-center"
        >
          Günəş
        </h6>
        <div
          class="col-5 col-md-12"
        >
          <h4>
            -:-
          </h4>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(246, 153, 63);"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm0-10c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="col-12 col-md-4 alert alert-success"
    >
      <div
        class="row"
      >
        <h6
          class="col-7 col-md-12 align-self-center"
        >
          Zöhr namazı
        </h6>
        <div
          class="col-5 col-md-12"
        >
          <h4>
            -:-
          </h4>
          <div
            class="col-12 text-center"
          >
            <span
              class="badge bg-warning text-dark"
            >
              Zaman varkən
              <br />
              Namazını qıl
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-12 col-md-4 alert"
    >
      <div
        class="row"
      >
        <h6
          class="col-7 col-md-12 align-self-center"
        >
          Əsr namazı
        </h6>
        <div
          class="col-5 col-md-12"
        >
          <h4>
            -:-
          </h4>
          <small>
            4 minutes ago
          </small>
        </div>
      </div>
    </div>
    <div
      class="col-12 col-md-4 alert"
    >
      <div
        class="row"
      >
        <h6
          class="col-7 col-md-12 align-self-center"
        >
          Məğrib namazı
        </h6>
        <div
          class="col-5 col-md-12"
        >
          <h4>
            -:-
          </h4>
          <small>
            5 minutes ago
          </small>
        </div>
      </div>
    </div>
    <div
      class="col-12 col-md-4 alert"
    >
      <div
        class="row"
      >
        <h6
          class="col-7 col-md-12 align-self-center"
        >
          İşa namazı
        </h6>
        <div
          class="col-5 col-md-12"
        >
          <h4>
            -:-
          </h4>
          <small>
            6 minutes ago
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;
